### PR TITLE
Prepare release v1.21.0

### PR DESCRIPTION
**Pull request by garkhipov**
Updates version and changelog for v1.21.0.

### CODE REVIEW

**Review by JacobOaks — COMMENTED** [review]
Hey! Thanks for the PR, I think it's a good time for this release. Looks like we haven't done a great job keeping our changelog up-to-date 😅. Can we add these items for this release entry (under "Changed")? 

* Upgrade Dig dependency to v1.17.1
* Update and clarify a lot of the API documentation

**Comment by JacobOaks** [comment]
#1172 should fix the lint issue in CI.

**Comment by garkhipov** [comment]
@JacobOaks extra lines added 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,11 +10,16 @@ All notable changes to this project will be documented in this file.
...
+## [1.21.0](https://github.com/uber-go/fx/compare/v1.20.1...v1.21.0) - 2024-03-13
 
 ### Added
 - fxtest: Add WithTestLogger option that uses a `testing.TB` as the
   Fx event logger.
+- An fxevent logger that can log events using a slog logger has been added.
+
+### Changed
+- Upgrade Dig dependency to v1.17.1
+- Update and clarify a lot of the API documentation
```
Comment (sywhang) [inline CHANGELOG.md:22]: Can we remove this? We should only contain user facing functional changes in the changelog.

**Review by tchung1118 — APPROVED** [review]
LGTM

**Review by sywhang — APPROVED** [review]

**Pull request merged**